Menus now work in briefing screen

Nomas-X opened the pull request:
All menus other of the progress bar now work in the briefing screen, this should allow mission makers to use them there should they need it.

Nomas-X committed:
Menus now work in briefing screen

Nomas-X committed:
Fixed client state condition check

ConnorAU commented:
Do the menus show if the briefing display is provided as the parent display argument?

Nomas-X commented:
They do but they end up hiding the briefing to show, video provided shows example:
https://www.youtube.com/watch?v=w7Z_tf9lVbI

The PR I made makes it so the menus show up over the briefing map.

Nomas-X commented:
> Do the menus show if the briefing display is provided as the parent display argument?

They do but they end up hiding the briefing to show, video provided shows example:
https://www.youtube.com/watch?v=w7Z_tf9lVbI

The PR I made makes it so the menus show up over the briefing map.

ConnorAU commented:
Okay fair enough. 

Is it possible this edit can be simplified to one command instead of using an if statement, or is there a reason not to use the alt syntax when not in briefing?

```sqf
createDialog [QUOTE(DISPLAY_NAME), getClientStateNumber == 8];
```

Nomas-X commented:
> Okay fair enough. 
> 
> Is it possible this edit can be simplified to one command instead of using an if statement, or is there a reason not to use the alt syntax when not in briefing?
> 
> ```sqf
> createDialog [QUOTE(DISPLAY_NAME), getClientStateNumber == 8];
> ```

Ir should be fine to simplfy, I mainly used an if statement as it was the first thing to come to mind.

Will update the PR today once I am awake.

Nomas-X committed:
Condition improvements for briefing check

Nomas-X commented:
> Okay fair enough.
> 
> Is it possible this edit can be simplified to one command instead of using an if statement, or is there a reason not to use the alt syntax when not in briefing?
> 
> ```sqf
> createDialog [QUOTE(DISPLAY_NAME), getClientStateNumber == 8];
> ```

Commit done, PR no longer includes the if statement.

Nomas-X committed:
Add check for client state number 9

- I added a check for `getClientStateNumber == 9` so they can also be called through the diaries if need be.

Nomas-X commented:
Additionally added a check for client state number 9. Now it also works if you try to call it through a diary execute command, which was my initial goal.

In case you are wondering client state number 10 is when the player is loaded in-game, 8 and 9 are when he is in briefing screen and diary.
